modulemd/v2/include/modulemd-2.0/modulemd-module-index.h

```diff
@@ -71,6 +71,35 @@ G_DECLARE_FINAL_TYPE (
   ModulemdModuleIndex, modulemd_module_index, MODULEMD, MODULE_INDEX, GObject)
 
 
+/**
+ * ModulemdReadHandler:
+ * @data: (inout): A private pointer to the data being read.
+ * @buffer: (out): The buffer to write the data from the source.
+ * @size: (in): The size of the buffer.
+ * @size_read: (out): The actual number of bytes read from the source.
+ *
+ * The prototype of a read handler.
+ *
+ * The read handler is called when the parser needs to read more bytes from the
+ * source. The handler should write not more than @size bytes to the @buffer.
+ * The number of written bytes should be set to the @size_read variable.
+ *
+ * This handler is identical to a
+ * [yaml_read_handler_t](https://github.com/yaml/libyaml/blob/master/include/yaml.h#L988)
+ * but is included here to avoid depending on yaml.h in modulemd headers.
```

Comment:
To be clear, this is about preventing the exposure of libyaml interfaces as an effect of including the libmodulemd headers, right?

Reply:
Yes. It's mostly to avoid needing to have `#include <yaml.h>` in public headers and thereby require `libyaml-devel` to be installed when developing against `libmodulemd-devel`.

Reply:
👍